packages/components/src/components/domainproperties/dataset/DatasetColumnMappingPanel.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2020 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import { Col, Row } from 'react-bootstrap';
+
+import { getServerContext } from '@labkey/api';
+
+import { SectionHeading } from '../SectionHeading';
+import { DomainFieldLabel } from '../DomainFieldLabel';
+import { DomainField, SelectInput } from '../../..';
+
+import { DatasetModel } from './models';
+import { getStudySubjectProp, getStudyTimepointLabel } from './actions';
+
+interface Props {
+    model: DatasetModel;
+    onColumnMappingChange: (participantIdField?: string, timePointField?: string) => void;
+    subjectColumnName: string;
+    timepointType: string;
+}
+
+interface State {
+    closestParticipantIdField?: string;
+    closestTimepointField?: string;
+}
+
+export class DatasetColumnMappingPanel extends React.PureComponent<Props, State> {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            closestParticipantIdField: this.findClosestColumn(this.props.subjectColumnName),
+            closestTimepointField: this.findClosestColumn(this.props.timepointType),
+        };
+
+        this.props.onColumnMappingChange(this.state.closestParticipantIdField, this.state.closestTimepointField);
+    }
+
+    compareNames(name1: string, name2: string): boolean {
+        return name1 === name2 || name1.indexOf(name2) >= 0 || name2.indexOf(name1) >= 0;
+    }
+
+    areColumnNamesAndTypesEquivalent(targetColumnName: string, inferredField: DomainField): boolean {
+        const { subjectColumnName, timepointType } = this.props;
+
+        const targetColumn = targetColumnName.toLowerCase().replace(' ', '');
+        const inferredFieldName = inferredField.name.toLowerCase().replace(' ', '');
+
+        if (targetColumnName === subjectColumnName) {
+            return this.compareNames(targetColumn, inferredFieldName) || inferredField.rangeURI === 'xsd:string';
```

Comment:
@labkey-ankurj Your last commit change the `&&` to `||` in this statement, was that intentional? Wouldn't that mean that this would match the first string field regardless of the field name?